tools/topology/topology2/include/pipelines/cavs/dai-copier-eqiir-module-copier-capture.conf

```diff
@@ -0,0 +1,126 @@
+#
+# BE capture pipeline: dai-copier-eqiir-module-copier-capture
+#
+# All attributes defined herein are namespaced
+# by alsatplg to "Object.Pipeline.dai-copier-eqiir-module-copier-capture.N.attribute_name"
+#
+# Usage: dai-copier-eqiir-module-copier-capture pipeline object can be instantiated as:
+#
+# Object.Pipeline.dai-copier-eqiir-module-copier-capture."N" {
+# 	period		1000
+# 	time_domain	"timer"
+# 	channels	2
+# 	rate		48000
+# }
+#
+# Where N is the unique pipeline ID within the same alsaconf node.
+#
+
+<include/common/audio_format.conf>
+<include/components/copier.conf>
+<include/controls/bytes.conf>
+<include/components/eqiir.conf>
+<include/components/pipeline.conf>
+
+Class.Pipeline."dai-copier-eqiir-module-copier-capture" {
+
+	DefineAttribute."index" {}
+
+	<include/pipelines/pipeline-common.conf>
+
+	attributes {
+		!constructor [
+			"index"
+		]
+
+		!immutable [
+			"direction"
+		]
+
+		#
+		# dai-copier-eqiir-module-copier-capture objects instantiated
+		# within the same alsaconf node must have unique pipeline_id
+		# attribute
+		#
+		unique	"instance"
+	}
+
+	Object.Widget {
+		copier."1" {
+			type dai_out
+			period_sink_count 2
+			period_source_count 0
+			num_audio_formats 1
+			num_input_audio_formats 1
+			num_output_audio_formats 1
+			# copier only supports one format based on mixin/mixout requirements: 32-bit 48KHz 2ch
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+		}
+
+		eqiir."1" {
+			num_audio_formats 1
+			num_input_audio_formats 1
+			num_output_audio_formats 1
+
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+			}
+
+			Object.Control.bytes."1" {
+				IncludeByKey.DMIC0_DAI_EQIIR {
+					"passthrough"		"include/components/eqiir/passthrough.conf"
+					"highpass_40hz_0db"	"include/components/eqiir/highpass_40hz_0db_48khz.conf"
+					"highpass_40hz_20db"	"include/components/eqiir/highpass_40hz_20db_48khz.conf"
+				}
+			}
+		}
+
+		copier."2" {
+			type buffer
+			copier_type	"module"
+			period_sink_count 2
+			period_source_count 0
+			num_audio_formats 1
+			num_input_audio_formats 1
+			num_output_audio_formats 1
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+		}
+
+		pipeline."1" {
+			priority	0
+			lp_mode		0
+		}
+	}
+
+	Object.Base {
+		route.1 {
+			source	eqiir.$index.1
+			sink	copier.module.$index.2
+		}
+	}
+
+	direction	"capture"
+	dynamic_pipeline 1
+	time_domain	"timer"
+	channels	2
+	channels_min	2
+	channels_max	2
```

Comment:
@RanderWang Did you mean this? I found yesterday that 4 channels capture worked despite of these set to 2. But it makes sense, this should be 4. Is changing only channels_max enough?

Reply:
I am not sure, but it is very possible.

Reply:
OK, I just fixed the channels_max